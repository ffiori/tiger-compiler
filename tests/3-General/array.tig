let
    type intArray = array of int
    var row := intArray [ 4 ] of 0
    function oper(i: int) : int = (
      printInt(100+i);
      if i > 0 then
        (row[i]:=i;
        oper(i-1);
        printInt(200+i);
        0)
      else 0 )
    in oper(3); 
       printInt(row[0]);
       printInt(row[1]);
       printInt(row[2]);
       printInt(row[3]);
       printInt(row[4]); /* should fail */
       0
end
