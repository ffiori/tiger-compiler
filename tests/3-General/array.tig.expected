INT: 103
INT: 102
INT: 101
INT: 100
INT: 201
INT: 202
INT: 203
INT: 0
INT: 1
INT: 2
INT: 3
indice 4 excedido!
